ctrl/ctrl.c

```diff
@@ -0,0 +1,276 @@
+
+/*
+ * Copyright (c) 2025 Pantacor Ltd.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "ctrl.h"
+#include "ctrl_util.h"
+#include "ctrl_caller.h"
+#include "ctrl_endpoints.h"
+#include "utils/list.h"
+#include "event/event_http_server.h"
+#include "init.h"
+
+#include <event2/http.h>
+#include <event2/buffer.h>
+
+#include <linux/limits.h>
+#include <string.h>
+#include <stdio.h>
+
+#define MODULE_NAME "ctrl"
+#define pv_log(level, msg, ...) vlog(MODULE_NAME, level, msg, ##__VA_ARGS__)
+#include "log.h"
+
+#define PV_CTRL_ERROR_RSP "{\"Error\": \"Unsupported protocol\"}"
+
+struct ctrl_srv {
+	struct evhttp *srv;
+	struct dl_list custom_cb;
+	struct dl_list normal_cb;
+};
+
+static struct ctrl_srv pvctrl = { 0 };
+
+// catch any error in the communication, for example use any other protocol
+// than http against the socket
+static int ctrl_error_cb(struct evhttp_request *req, struct evbuffer *buf,
+			 int err, const char *reason, void *ctx)
+{
+	(void)req;
```

Comment:
I have to insist with the logs for adding and executing callbacks... it is surely verbose, but we have those in master now, then we can move them to a lower log level, or under PV_LIBEVENT_DEBUG_MODE. Anything could be done, but we should discuss it as a team and think about code and logs as a whole

Reply:
And you could say: "there are too many logs!", but the real problem is "there are too many events!", so we should cut on events, not logs, and logs are good for flagging when events are happening.